src/ui/templates/insurance/your-business/company-details.njk

```diff
@@ -98,109 +98,154 @@
-      {{ govukRadios({
-        classes: "govuk-radios--inline",
-        name: "trading-name",
-        fieldset: {
-          legend: {
-            text: CONTENT_STRINGS.TRADING_NAME,
-            isPageHeading: false,
-            classes: "govuk-fieldset__legend--m"
-          },
-          attributes: {
-            'data-cy': 'trading-name'
-          }
-        },
-        items: [
-          {
-            value: "yes",
-            text: "Yes",
+    <form action="{{COMPANY_DETAILS_POST_ROUTE}}" method="POST" novalidate>
+      <input type="hidden" name="_csrf" value="{{ csrfToken }}">
+
+      <div class="govuk-grid-column-three-quarters-from-desktop">
+        {{ govukRadios({
+          classes: "govuk-radios--inline",
+          idPrefix: FIELDS.YOUR_COMPANY.TRADING_NAME,
+          name: FIELDS.YOUR_COMPANY.TRADING_NAME,
+          fieldset: {
+            legend: {
+              text: CONTENT_STRINGS.TRADING_NAME,
+              isPageHeading: false,
+              classes: "govuk-fieldset__legend--m"
+            },
             attributes: {
-              'data-cy': "trading-name-yes"
+              'data-cy': 'trading-name'
             }
           },
-          {
-            value: "no",
-            text: "No",
+          items: [
+            {
+              value: "yes",
+              text: "Yes",
+              checked: submittedValues[FIELDS.YOUR_COMPANY.TRADING_NAME] === 'yes',
+              attributes: {
+                'data-cy': "trading-name-yes",
+                'aria-label': CONTENT_STRINGS.TRADING_NAME + ' yes radio'
+              }
+            },
+            {
+              value: "no",
+              text: "No",
+              checked: submittedValues[FIELDS.YOUR_COMPANY.TRADING_NAME] === 'no',
+              attributes: {
+                'data-cy': "trading-name-no",
+                'aria-label': CONTENT_STRINGS.TRADING_NAME + ' no radio'
+              }
+            }
+          ],
+          errorMessage: validationErrors.errorList.tradingName and {
+            text: validationErrors.errorList.tradingName.text,
             attributes: {
-              'data-cy': "trading-name-no"
+              "data-cy": "trading-name-error"
             }
           }
-        ]
-      }) }}
-    </div>
+        }) }}
+      </div>
 
-    <div class="govuk-grid-column-three-quarters-from-desktop">
-      {{ govukRadios({
-        classes: "govuk-radios--inline",
-        name: "trading-address",
-        fieldset: {
-          legend: {
-            text: CONTENT_STRINGS.TRADING_ADDRESS,
+      <div class="govuk-grid-column-three-quarters-from-desktop">
+        {{ govukRadios({
+          classes: "govuk-radios--inline",
+          name: FIELDS.COMPANY_DETAILS.TRADING_ADDRESS,
+          fieldset: {
+            legend: {
+              text: CONTENT_STRINGS.TRADING_ADDRESS,
+              isPageHeading: false,
+              classes: "govuk-fieldset__legend--m"
```

Comment:
Do we have an E2E test for the radio label/legend? Bit of faff adding `data-cy` to `govukRadios` because it doesn't support `legend.attributes` so have to type out the html instead. This is already done in the `yesNoRadioButtons` component.

Might need to tweak the component to generate custom `data-cy` value for the legend. 

Could you have a look? Not sure if it's worth the extra workright now? Could be tech improvement ticket. The component is currently only used in "yes or no" answer pages but this page is a little more complex